### PR TITLE
API-21862 mpi user endpoint

### PR DESCRIPTION
**Pull request by crolarlibertyva**
Addresses:
-  https://vajira.max.gov/browse/API-21016
-  https://vajira.max.gov/browse/API-21862

Changes:
- VetsApiClient replaced by MpiUserClient and VsoClient
- SAMLUser broken out into separate interface

Remaining items:
- [x] Rerun regression tests when ID.me is functional
- [x] Config examples update
- [x] README update
- [x] Param store updates 
- [x]  config PR merge.
  - https://github.com/department-of-veterans-affairs/lighthouse-saml-proxy-configs/pull/37

### CODE REVIEW

File: src/MpiUserClient.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    const body: Record<string, any> = {
+      idp_uuid: user.uuid,
+      dslogon_edipi: user.edipi || null,
+      mhv_icn: user.icn || null,
+      ssn: user.ssn || null,
+      first_name: user.firstName || null,
+      middle_name: user.middleName || null,
+      last_name: user.lastName || null,
+      dob: user.dateOfBirth || null,
+      gender: user.gender?.substring(0, 1).toUpperCase() || null,
```
Comment (dbrown-va) [inline src/MpiUserClient.ts:28]: I didn't realize that was something the vets-api side was providing.

But sure enough..
`gender: user_attributes[:gender]&.chars&.first&.upcase`

The more you know.

File: src/MpiUserClient.test.ts [hunk 2]
```diff
@@ -55,72 +54,85 @@ beforeEach(() => {
...
-    await client.getMVITraitsForLoa3User(samlTraitsICN);
+    const client = new MpiUserClient(
+      "faketoken",
+      "https://example.gov//mvi-user",
+      "faketoken"
+    );
+    await client.getMpiTraitsForLoa3User(samlTraitsICN);
     expect(request.post).toHaveBeenCalledWith({
-      url: "https://example.gov/internal/auth/v0/mvi-user",
+      url: "https://example.gov//mvi-user",
```
Comment (johnmweisz) [inline src/MpiUserClient.test.ts:90]: I see double /

File: dev-config.base.json [hunk 3]
```diff
@@ -17,7 +17,9 @@
   "spButtonText": "Sign in",
   "idpEncryptionCert": "./idp-encryption-cert.pem",
   "idpEncryptionPublicKey": "./idp-encryption-pub.key",
-  "vetsAPIHost": "https://vets-api.example.com",
+  "mpiUserEndpont": "https://dev.example.com/mpiuser",
+  "vsoUserEndpont": "https://dev.example.com/vsouser",
```
Comment (johnmweisz) [inline dev-config.base.json:21]: what is vso?

File: src/MpiUserClient.test.ts [hunk 4]
```diff
@@ -55,72 +54,85 @@ beforeEach(() => {
...
       }),
     });
   });
 
   it("should call the mvi-user endpoint with the Veteran's PII in request body", async () => {
-    const client = new VetsAPIClient("faketoken", "https://example.gov");
-    await client.getMVITraitsForLoa3User(samlTraits);
+    const client = new MpiUserClient(
+      "faketoken",
+      "https://example.gov/mpi-user",
```
Comment (johnmweisz) [inline src/MpiUserClient.test.ts:106]: should this include /auth?

Comment (johnmweisz) [reply src/MpiUserClient.test.ts:106]: I guess its just a placeholder?

**Review by johnmweisz — APPROVED** [review]

**Pull request merged**